File: Data Structures & Algorithms/trapping-rain-water/submission-1.java
class Solution {
    public int trap(int[] h) {
        int[] leftM = new int[h.length];
        int[] rightM = new int[h.length];
        int max = 0;
        for (int i = 1; i < h.length; i ++) {
            max = Math.max(h[i-1], max);
            leftM[i] = max;
        }
        max = 0;
        for (int i = h.length - 2; i >= 0; i --) {
            max = Math.max(h[i+1], max);
            rightM[i] = max;
        }
        int ans = 0;
        for (int i = 1; i < h.length - 1; i ++) {
            int cur = Math.min(leftM[i], rightM[i]) - h[i];
            if (cur > 0) ans += cur;
        }
        return ans;
    }
}

File: Data Structures & Algorithms/merge-intervals/submission-1.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x:x[0])
        ans = [intervals[0]]
        def overlap(i1, i2):
            return not (i1[0]>i2[1] or i2[0]>i1[1])

        for i, interval in enumerate(intervals):
            if i == 0:
                continue
            if overlap(ans[-1], interval):
                ans[-1][0] = min(ans[-1][0], interval[0])
                ans[-1][1] = max(ans[-1][1], interval[1])
            else:
                ans.append(interval)
        
        return ans